xbmc/utils/SeekHandler.cpp

```diff
@@ -90,45 +90,42 @@ void CSeekHandler::Reset()
   }
 }
 
-int CSeekHandler::GetSeekSeconds(bool forward, SeekType type)
+int CSeekHandler::GetSeekStepSize(SeekType type, int step)
 {
-  m_seekStep = m_seekStep + (forward ? 1 : -1);
-
-  if (m_seekStep == 0)
-    return 0;
-
-  std::vector<int> seekSteps(m_seekStep > 0 ? m_forwardSeekSteps.at(type) : m_backwardSeekSteps.at(type));
+  std::vector<int> seekSteps(step > 0 ? m_forwardSeekSteps.at(type) : m_backwardSeekSteps.at(type));
 
   if (seekSteps.empty())
   {
     CLog::Log(LOGERROR, "SeekHandler - %s - No %s %s seek steps configured.", __FUNCTION__,
-              (type == SeekType::SEEK_TYPE_VIDEO ? "video" : "music"), (m_seekStep > 0 ? "forward" : "backward"));
+              (type == SeekType::SEEK_TYPE_VIDEO ? "video" : "music"), (step > 0 ? "forward" : "backward"));
     return 0;
   }
 
   int seconds = 0;
 
   // when exceeding the selected amount of steps repeat/sum up the last step size
-  if (static_cast<size_t>(abs(m_seekStep)) <= seekSteps.size())
-    seconds = seekSteps.at(abs(m_seekStep) - 1);
+  if (static_cast<size_t>(abs(step)) <= seekSteps.size())
+    seconds = seekSteps.at(abs(step) - 1);
   else
-    seconds = seekSteps.back() * (abs(m_seekStep) - seekSteps.size() + 1);
+    seconds = seekSteps.back() * (abs(step) - seekSteps.size() + 1);
 
   return seconds;
 }
 
 void CSeekHandler::Seek(bool forward, float amount, float duration /* = 0 */, bool analogSeek /* = false */, SeekType type /* = SEEK_TYPE_VIDEO */)
 {
-  // abort if we do not have a play time or already perform a seek
-  if (m_performingSeek)
-    return;
-
   CSingleLock lock(m_critSection);
 
   // not yet seeking
   if (!m_requireSeek)
   {
-    // tell info manager that we have started a seek operation
+    // use only the first step forward/backward for a seek without a delay
+    if (!analogSeek && m_seekDelays.at(type) == 0)
+    {
+      SeekSeconds(GetSeekStepSize(type, forward ? 1 : -1));
```

Comment:
just a brief look during a break here. I don't think we should hold the lock here